File: py/linkedlists/linkedlists.py
class SinglyLinkedList:
    val: int
    next = None

    def __init__(self, val: int):
        self.val = val
        self.next = None

    @staticmethod
    def from_arr(arr: list[int]):
        head = SinglyLinkedList(arr[0])
        curr = head
        i = 1
        while i < len(arr):
            curr.next = SinglyLinkedList(arr[i])
            curr = curr.next
            i += 1
        return head


class DoublyLinkedList:
    val: int
    prev = None
    next = None

    def __init__(self, val: int):
        self.val = val
        self.next = None

    @staticmethod
    def from_arr(arr: list[int]):
        head = DoublyLinkedList(arr[0])
        curr = head
        i = 1
        while i < len(arr):
            curr.next = DoublyLinkedList(arr[i])
            curr = curr.next
            i += 1
        return head

File: go/linkedlists/linkedlists.go
package linkedlists

type SinglyLinkedNode struct {
	Val  int
	Next *SinglyLinkedNode
}

type DoublyLinkedNode struct {
	Val  int
	Prev *DoublyLinkedNode
	Next *DoublyLinkedNode
}

func arrToSinglyLinkedList(nums []int) *SinglyLinkedNode {
	var (
		head                   = SinglyLinkedNode{nums[0], nil}
		curr *SinglyLinkedNode = &head
	)
	for i := 1; i < len(nums); i++ {
		curr.Next = &SinglyLinkedNode{nums[i], nil}
		curr = curr.Next
	}
	return &head
}

func arrToDoublyLinkedList(nums []int) *DoublyLinkedNode {
	var (
		head                   = DoublyLinkedNode{nums[0], nil, nil}
		curr *DoublyLinkedNode = &head
	)
	for i := 1; i < len(nums); i++ {
		curr.Next = &DoublyLinkedNode{
			Val:  nums[i],
			Prev: curr,
			Next: nil,
		}
		curr = curr.Next
	}
	return &head
}
